src/templates/vue/src/components/NetworkStatus.vue

```diff
@@ -0,0 +1,110 @@
+<template>
+  <div>
+    <div v-if="!online" class="msg offline-msg">
+        <div class="icon">
+          <span></span>
+        </div>
+        <div class="content">          
+          <div><strong>Offline : </strong>Connection lost! You are not connected to internet</div>
+        </div>
+    </div>
+    <div v-if="isBackOnline" class="msg online-msg">
+        <div class="icon">
+          <img src="https://img.icons8.com/office/40/000000/high-connection.png"/>
+        </div>
+        <div class="content">
+          <div><strong>Online : </strong>Back to online</div>
+        </div>
+    </div>
+  </div>
+</template>
+
+<script>
+const ONLINE = "online"
+const OFFLINE = "offline"
+
+export default {
+  name : "NetworkStatus",
+  data() {
+    return {
+      online: navigator.onLine,
+      isBackOnline: false,
+    };
+  },
+  methods: {
+    updateOnlineStatus(event) {
+      this.online = event.type === ONLINE;
+    },
+  },
+  watch: {
+    online(value) {
+      if (value) {
+        this.isBackOnline = true;
+        setTimeout(() => {
+          this.isBackOnline = false;
+        }, 1000);
+      }
+    },
+  },
+  mounted() {
+    window.addEventListener(ONLINE, this.updateOnlineStatus);
+    window.addEventListener(OFFLINE, this.updateOnlineStatus);
+  },
+  beforeDestroy() {
+    window.removeEventListener(ONLINE, this.updateOnlineStatus);
+    window.removeEventListener(OFFLINE, this.updateOnlineStatus);
+  },
+};
+</script>
+
+<style>
+.msg{
+  display: flex;
+  align-items: center;
+  text-align: left;
+  padding-top: 0.4rem;
+  padding-bottom: 0.4rem;
+  font-size: 0.75rem;
+}
+.offline-msg{
+  background: #fee8ec !important;
```

Comment:
we should use a themes.js file and get colors from there
so that if we have to change theme, we dont have to change colors from multiple places

Reply:
kept in the different CSS file but there is no effect of theme on the Network status prompt